samples/drivers/rtc/sample.yaml

```diff
@@ -6,6 +6,15 @@ tests:
       - stm32f3_disco
       - mimxrt700_evk/mimxrt798s/cm33_cpu0
       - mimxrt700_evk/mimxrt798s/cm33_cpu1
+      - frdm_k22f
```

Comment:
those platforms share the same config, maybe you can use a 
```
sample.drivers.nxp_rtc
    filter: dt_compat_enabled("nxp-rtc")
    extra_configs:
      - CONFIG_RTC_ALARM=y
      - CONFIG_RTC_UPDATE=y
      - CONFIG_RTC_CALIBRATION=y
```
and move below to common
```
common:
    tags:
      - samples
      - rtc
      - api
    depends_on:
      - rtc
    harness: console
    harness_config:
      type: one_line
      regex:
        - "RTC date and time: 2024-11-17 04:19:01"
```
and change the origin on with another filter
```
sample.drivers.rtc:
  filter: not dt_compat_enabled("nxp-rtc")
```